### PR TITLE
Ensure that META-INF need is clear in the docs

### PR DESCRIPTION
**Pull request by nmldiegues**
This merely updates the README to clarify the need for the META-INF as mentioned in https://github.com/feedzai/feedzai-openml/issues/17

### CODE REVIEW

**Comment by nmldiegues** [comment]
@pedrorijo91  please take a look for https://github.com/feedzai/feedzai-openml/issues/17

File: README.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ The key concept is the `MachineLearningProvider`. To implement an OpenML provide
 
 * To train new machine learning algorithms within the Feedzai platform, implement the `TrainingMachineLearningProvider` extension of the `MachineLearningProvider` interface.
 
+* Finally, generate a Jar with the code (possibly with all the dependencies necessary in it, or a set of Jars instead), and make sure you include META-INF to indicate which providers you have in your code. Check out [our example in this repository](https://github.com/feedzai/feedzai-openml/blob/master/openml-example/src/main/resources/META-INF/services/com.feedzai.mlapi.provider.MachineLearningProvider).
```
Comment (pedrorijo91) [inline README.md]: I think we are lacking the explanation as why would someone need to add the META-INF on the jar file. While it's an implementation detail, we can't hide that fact from OpenML developers, and as such I feel we own them the explanation that the Feedzai platform (pulse) will look for any provider through a java ServiceLoader, hence the need for this declaration file

Comment (nmldiegues) [reply README.md]: I already say that this is to "make sure ... indicate which providers you have". That should be more than enough without getting into dirty details, no?

Comment (pedrorijo91) [reply README.md]: true, I can live with hiding those details (that we are using a ServiceLoader and so on). But I truly believe that we should be more specific on this point: There should be a file under `resources/META-INF/services/com.feedzai.openml.MachineLearningProvider` (not sure if it has to be under resources?) with the fully qualified name of all providers, one per line

Comment (henriquevcosta) [reply README.md]: Honestly ServiceLoader is such a well known standard/mechanism that I think it's easier to just to link to that and say that the service interface is `com.feedzai.openml.MachineLearningProvider` than to have to describe that it is a file in META-INF with name x and contents y...

Comment (JPDSousa) [reply README.md]: Just a side note here: Google's [Auto Service](https://github.com/google/auto/tree/master/service) uses annotation processing to generate the META-INF resources. IMO it's a good tip for people who are not used to dealing with java's Service Loader.

**Review by pedrorijo91 — APPROVED** [review]

**Comment by pedrorijo91** [comment]
cherry-pick into hf-0.1.X

**Pull request merged**